src/Storages/transformQueryForExternalDatabase.cpp

```diff
@@ -306,6 +306,18 @@ String transformQueryForExternalDatabase(
         throw Exception("Query contains non-compatible expressions (and external_table_strict_query=true)", ErrorCodes::INCORRECT_QUERY);
     }
 
+    auto * literal_expr = typeid_cast<ASTLiteral *>(original_where.get());
+    UInt64 value;
+    if (literal_expr && literal_expr->value.tryGet<UInt64>(value) && (value == 0 || value == 1))
+    {
+        /// WHERE 1 -> WHERE 1=1, WHERE 0 -> WHERE 1=0.
```

Comment:
Ok, but what about, for example, `WHERE 42`?

Reply:
It will also work with this code (checked). Because at this point `WHERE expr=constant` will be either `0` or `1`.